From 3b4990d5042c9e6520fa1bd6ff2be696243b3792 Mon Sep 17 00:00:00 2001
From: Giovanni Campagna <gcampagn@redhat.com>
Date: Fri, 13 Sep 2013 15:52:30 +0200
Subject: [PATCH 25/39] Fix fallback loading of the wayland driver

Fix the array name to actually compile, and fix the driver name
with the new upstream.
---
 hw/xfree86/common/xf86AutoConfig.c | 2 +-
 hw/xfree86/common/xf86Config.c     | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/hw/xfree86/common/xf86AutoConfig.c b/hw/xfree86/common/xf86AutoConfig.c
index d5a3370..b35cb96 100644
--- a/hw/xfree86/common/xf86AutoConfig.c
+++ b/hw/xfree86/common/xf86AutoConfig.c
@@ -285,7 +285,7 @@ listPossibleVideoDrivers(char *matches[], int nmatches)
     if (i < (nmatches - 1)) {
 #ifdef XORG_WAYLAND
         if (xorgWayland)
-            matches[i++] = xnfstrdup("wlshm");
+            matches[i++] = xnfstrdup("wayland");
         else
 #endif
 #if !defined(__linux__) && defined(__sparc__)
diff --git a/hw/xfree86/common/xf86Config.c b/hw/xfree86/common/xf86Config.c
index c65cb80..8255d8d 100644
--- a/hw/xfree86/common/xf86Config.c
+++ b/hw/xfree86/common/xf86Config.c
@@ -521,7 +521,7 @@ fixup_video_driver_list(char **drivers)
 {
     static const char *fallback_hw[4] = { "vesa", "fbdev", "wsfb", NULL };
 #ifdef XORG_WAYLAND
-    static const char *fallback_wayland[2] = { "wlshm", NULL };
+    static const char *fallback_wl[2] = { "wayland", NULL };
 #endif
     const char **fallbacks;
     char **end, **drv;
-- 
1.8.3.1

